### PR TITLE
Add Reproducion logs for pyserini

### PR DESCRIPTION
**Pull request by ljk423**
#### Reproduction Log for start.md

**Date:** 2023-12-04

### Setup Details:
**Operating System** 

- Mac OS Monterey ver 12.2.1

**Software Environment**

- Python 3.10.13
- Java 11.0.13

**Hardware Environment**

- CPU: Apple M1
- RAM: 8GB
- GPU: Apple M1

**Success/Failure**

- Everything Worked

**Additional Note**

- For some reason, the import statement for CosDprDocumentEncoder in pyserini/encode/__init__.py was deleted, so I added it manually.
- During the indexing process in experiments-nfcorpus.md, a segmentation fault occurred occasionally on the Apple M1 chip. After debugging using faulthandler, it was confirmed to be a torch-related issue. The problem was resolved by uninstalling torch using conda remove and reinstalling it. It is suspected that there was a conflict with the existing installation of torch.

### CODE REVIEW

**Review by ljk423 — COMMENTED** [review]
Changed commit hash into main trunk's hash

**Comment by lintool** [comment]
Please fix conflicts.

**Comment by ljk423** [comment]
I fixed the conflicts. Sorry for inconvenience.
Thanks

**Comment by lintool** [comment]
There should only be diffs for the four repro files. You have a lot more extra files as part of this PR. Please fix.

**Comment by ljk423** [comment]
I apologize, Professor. I have just deleted/restored all the files that were recently added or modified. I apologize once again for any inconvenience this may have caused.

Thanks

File: pyserini/encode/__init__.py [hunk 1]
```diff
@@ -27,4 +27,4 @@
 from ._splade import SpladeQueryEncoder
 from ._slim import SlimQueryEncoder
 from ._openai import OpenAIDocumentEncoder, OpenAIQueryEncoder, OPENAI_API_RETRY_DELAY
-from ._cosdpr import CosDprEncoder, CosDprDocumentEncoder, CosDprQueryEncoder
```
Comment (lintool) [inline pyserini/encode/__init__.py:30]: Is this change needed? Please undo.

**Review by lintool — APPROVED** [review]

**Review by lintool — CHANGES_REQUESTED** [review]
You need to also remove `docs/collections/nfcorpus.zip`

**Comment by ljk423** [comment]
I just deleted the collections directory in docs
I'm really sorry for inconvenience

**Pull request merged**